### PR TITLE
use consistent unique value for staticfiles cache prefix

### PR DESCRIPTION
**Pull request by thraxil**
With the hostname+random number approach, every edxapp server gets a different KEY_PREFIX. Eg, juniper prod currently has:

`/edx/etc/lms.yml` on edxapp0 and edxapp1:

```
KEY_PREFIX: prod-tahoe-us-juniper-edxapp-0_2666298_general

KEY_PREFIX: prod-tahoe-us-juniper-edxapp-1_5857041_general
```

`/edx/etc/studio.yml` on edxapp0 and edxapp1:

```
KEY_PREFIX: prod-tahoe-us-juniper-edxapp-0_4594680_general

KEY_PREFIX: prod-tahoe-us-juniper-edxapp-1_14407_general
```

etc. That means that they are not sharing the cache. So each server has to do the URL generation/lookup for itself the first time it gets the first request for a given static file.

Instead, we should be using the same `KEY_PREFIX` on all the servers, except we want to be very careful that it changes on each deploy so the cache is forced to update when staticfiles may have changed.

To do that, https://github.com/appsembler/tahoe_deploy/pull/42 passes in the Cloud Build `BUILD_ID` variable as
`EDXAPP_APPSEMBLER_BUILD_ID`. That should fit our requirement of being unique per deploy but consistent across servers.

### CODE REVIEW

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -1430,7 +1435,7 @@ generic_env_config:  &edxapp_generic_env
       <<: *default_generic_cache
       # Tahoe: RED-1961 Adding a random prefix to invalidate deployment on each deployment to avoid collision.
       #        9100100 is just a random big enough number.
-      KEY_PREFIX: "{{ ansible_hostname|default('staticfiles') }}_{{ 9100100 | random }}_general"
+      KEY_PREFIX: "{{ EDXAPP_APPSEMBLER_BUILD_ID }}_general"
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/defaults/main.yml]: I have no idea why it was `_general` but `staticfiles` makes more sense.

```suggestion
      KEY_PREFIX: "{{ EDXAPP_APPSEMBLER_BUILD_ID }}_staticfiles"
```

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -1430,7 +1435,7 @@ generic_env_config:  &edxapp_generic_env
       <<: *default_generic_cache
       # Tahoe: RED-1961 Adding a random prefix to invalidate deployment on each deployment to avoid collision.
       #        9100100 is just a random big enough number.
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/defaults/main.yml]: No longer needed.

```suggestion
```

**Review by OmarIthawi — APPROVED** [review]
Looks great! Thanks Anders.

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Sorry for the late minute request. the cache prefix has to be different in `lms` and `studio`. the randomness used to achieve that.

The following could work:

```yml
prefix: "staticfiles_{{ build_id }}_{{ service_variant }}"
```

I don't know how can we use `service_variant` though.

**Comment by thraxil** [comment]
@OmarIthawi yeah, I wasn't positive how that should work. I think if we need a different key for lms/cms, we should do that in `lms/envs/production.py` and `cms/envs/production.py` in the `edx-platform` code. Eg, we currently have: https://github.com/appsembler/edx-platform/blob/main/cms/envs/production.py#L212-L216

```
# if 'staticfiles' in CACHES:
#     CACHES['staticfiles']['KEY_PREFIX'] = EDX_PLATFORM_REVISION
```

I think what we want is to put that back in but with a change like:

```
if 'staticfiles' in CACHES:
    CACHES['staticfiles']['KEY_PREFIX'] = CACHES['staticfiles']['KEY_PREFIX'] + "_cms"
```

And maybe leave the lms one as it is. That way, cms and lms have independent caches, but are otherwise consistent and sharing across servers.

**Comment by OmarIthawi** [comment]
> @OmarIthawi yeah, I wasn't positive how that should work. I think if we need a different key for lms/cms, we should do that in `lms/envs/production.py` and `cms/envs/production.py` in the `edx-platform` code. Eg, we currently have: https://github.com/appsembler/edx-platform/blob/main/cms/envs/production.py#L212-L216
> 
> ```
> # if 'staticfiles' in CACHES:
> #     CACHES['staticfiles']['KEY_PREFIX'] = EDX_PLATFORM_REVISION
> ```
> 
> I think what we want is to put that back in but with a change like:
> 
> ```
> if 'staticfiles' in CACHES:
>     CACHES['staticfiles']['KEY_PREFIX'] = CACHES['staticfiles']['KEY_PREFIX'] + "_cms"
> ```
> 
> And maybe leave the lms one as it is. That way, cms and lms have independent caches, but are otherwise consistent and sharing across servers.

That definitely works! Thanks Anders!

**Comment by thraxil** [comment]
https://github.com/appsembler/edx-platform/pull/920 will be required before merging this.

**Review by OmarIthawi — APPROVED** [review]
Thanks @thraxil! That's a great fix. Now we triple our cache utility.

**Pull request merged**